sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala

```diff
@@ -204,13 +194,21 @@ case class CreateDataSourceTableAsSelectCommand(
       case None => data
     }
 
+    val tableWithPath = if (table.tableType == CatalogTableType.MANAGED) {
+      table.withNewStorage(
+        locationUri = Some(sessionState.catalog.defaultTablePath(table.identifier)))
```

Comment:
I think we should add a method in CatalogTable to return the location Uri by combining database path + table name.